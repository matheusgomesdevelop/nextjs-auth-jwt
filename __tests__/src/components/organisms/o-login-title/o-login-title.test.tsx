import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import LoginTitle from "@/components/organism/o-login-title/LoginTitle";

describe("Deve renderizar o o-login-title, corretamente", () => {
  it("Deve renderizar o a-title", () => {
    render(<LoginTitle title="" />);

    const get_a_title = screen.getByTestId("a-title");

    expect(get_a_title).toBeInTheDocument();
  });

  it("Deve renderizar o a-text", () => {
    render(<LoginTitle title="" />);

    const get_a_text = screen.getByTestId("a-text");

    expect(get_a_text).toBeInTheDocument();
  });

  it("Deve preservar a estrutura visual do componente", () => {
    const get_login_title = renderer.create(<LoginTitle title="NextJS: Auth + JWT" />).toJSON();

    expect(get_login_title).toMatchSnapshot();
  });
});
