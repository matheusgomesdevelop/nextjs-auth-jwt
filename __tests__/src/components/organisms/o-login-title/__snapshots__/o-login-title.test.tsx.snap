// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o o-login-title, corretamente Deve preservar a estrutura visual do componente 1`] = `
<div
  className="o_loginTitle"
  data-testid="o-login-title"
>
  <i
    className="a_icon pi pi-lock"
  />
  <h2
    className="a_title__h2"
    data-testid="a-title"
  >
    NextJS: Auth + JWT
  </h2>
  <p
    className="a_text__fwReg_fs16_gray500"
    data-testid="a-text"
  >
    Estamos felizes em vÃª-lo novamente! Insira suas credenciais para entrar.
  </p>
</div>
`;
